File: demo_langs/invalid/sample_bad.ts
// TypeScript - invalid code (syntax error)
interface User {
    id: number;
    name: string  // brak średnika
}

class UserService {
    private users: User[] = []
    
    addUser(user: User) {  // brak typu zwracanego
        this.users.push(user
    }  // brak zamknięcia nawiasu
    
    getUser(id: number) User | undefined {  // brak =>
        return this.users.find(u => u.id === id)
    }
}

const service = new UserService()
service.addUser({ id: 1, name: "Alice" }  // brak zamknięcia nawiasu i średnika

File: demo_langs/invalid/sample_bad.rs
// Rust - invalid code
struct User {
    id u32,  // brak dwukropka
    name String,  // brak dwukropka
}

impl User {
    fn new(id: u32 name: &str) -> Self {  // brak przecinka
        Self {
            id
            name: name.to_string()  // brak przecinka
        }  // brak przecinka
    }
}

fn main() {
    let user = User::new(1 "Alice")  // brak przecinka i średnika
    println!("Hello"  // brak zamknięcia nawiasu i średnika
